frontend-react/e2e/spec/all/authenticated/admin/organization-receiver-message-test.spec.ts

```diff
@@ -0,0 +1,299 @@
+                await expect(organizationReceiverMessageTestPage.page).toHaveURL(
+                    organizationReceiverMessageTestPage.url,
+                );
+                await expect(organizationReceiverMessageTestPage.page).toHaveTitle(
+                    organizationReceiverMessageTestPage.title,
+                );
+            });
+
+            test("displays test messages", async ({ organizationReceiverMessageTestPage }) => {
+                await expect(organizationReceiverMessageTestPage.form).toBeVisible();
+                let i = 0;
+                for (const message of organizationReceiverMessageTestPage.testMessages) {
+                    i++;
+                    const option = organizationReceiverMessageTestPage.form.getByLabel(message.fileName);
+                    await expect(option).toBeVisible();
+                    await expect(option).toHaveValue(message.reportBody);
+                }
+                expect(organizationReceiverMessageTestPage.testMessages.length).toEqual(i);
+            });
+
+            test("can add custom message", async ({ organizationReceiverMessageTestPage }) => {
+                const message1 = '{"foo":"bar"}';
+                const message2 = '{"bar":"foo"}';
+
+                await expect(organizationReceiverMessageTestPage.form).toBeVisible();
+                await expect(organizationReceiverMessageTestPage.addCustomMessageButton).toBeVisible();
+                await expect(organizationReceiverMessageTestPage.submitButton).toBeVisible();
+
+                for (const [i, message] of [message1, message2].entries()) {
+                    const filename = `Custom message ${i + 1}`;
+                    const customOption = organizationReceiverMessageTestPage.form.getByLabel(filename);
+                    const customOptionViewButton = organizationReceiverMessageTestPage.form
+                        .getByText(filename)
+                        .getByRole("button");
+
+                    await organizationReceiverMessageTestPage.addCustomMessageButton.click();
+                    await expect(organizationReceiverMessageTestPage.customMessageTextArea).toBeVisible();
+                    await expect(organizationReceiverMessageTestPage.submitCustomMessageButton).toBeVisible();
+                    await expect(organizationReceiverMessageTestPage.cancelCustomMessageButton).toBeVisible();
+
+                    await organizationReceiverMessageTestPage.customMessageTextArea.fill(message);
+                    await expect(organizationReceiverMessageTestPage.customMessageTextArea).toHaveValue(message);
+                    await organizationReceiverMessageTestPage.submitCustomMessageButton.click();
+                    await expect(organizationReceiverMessageTestPage.customMessageTextArea).toBeHidden();
+                    await expect(organizationReceiverMessageTestPage.submitCustomMessageButton).toBeHidden();
```

Comment:
Might be worth checking that the `Submit` button should be unclickable if there's nothing in the textarea.